### ISSUE TITLE
Legend disappears when color map is changed

### ISSUE THREAD

**Issue by vctmohan** [K3D-tools/K3D-jupyter#246]
The legend seemed to be disappearing when then the color map is changed. I am not sure if this is intended behavior. The following code can be used to reproduce the problem.

### How to reproduce

- Execute the following command.

```
import numpy as np
import k3d

plot = k3d.plot(height=320)

vertices = np.array([
    [-10, 0, -1],
    [10, 0, -1],
    [10, 0, 1],
    [-10, 0, 1],
], dtype=np.float32)
indices = np.array([
    [0, 1, 3],
    [1, 2, 3],
], dtype=np.uint32)
vertex_attribute = [0, 1, 1, 0]

mesh = k3d.mesh(vertices, indices, attribute=vertex_attribute,
                color_map=k3d.basic_color_maps.Rainbow)
plot += mesh

plot.display()
```

- This produces the following plot.

![K3D-1602957047592](https://user-images.githubusercontent.com/54903857/96349774-76a91800-10cf-11eb-9a12-e587a7e75a2a.png)

- Now, change the color map using as shown below:

```
mesh.color_map = k3d.paraview_color_maps.Cool_to_Warm
```

- This changes the color map. But, this also makes the legend disappear as shown in the following snapshot.

![K3D-1602957225731](https://user-images.githubusercontent.com/54903857/96349868-fdf68b80-10cf-11eb-9be9-3d594a38721b.png)

### System configuration

- K3D version: master branch
- Jupyter notebook server: 6.1.4
- Python: 3.7.6
- IPython 7.18.1
- OS: Microsoft Windows 10

**Comment by artur-trzesiok** [comment]
Thanks a lot! Will be fixed in next release (2.9.1)

**Comment by artur-trzesiok** [comment]
Please update k3d to 2.9.1
